### PR TITLE
Respect felixconfig health port

### PR DESCRIPTION
**Pull request by caseydavenport**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/installation/core_controller.go [hunk 1]
```diff
@@ -1177,6 +1180,22 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 			nodeTerminating = true
 		}
 	}
+
+	// Determine the felix health port to use. Prefer the configuration from FelixConfiguration,
+	// but default to 9099.
+	felixHealthPort := 9099
+	if felixConfiguration.Spec.HealthPort != nil {
```
Comment (tmjd) [inline pkg/controller/installation/core_controller.go:1187]: I think you're still probably working/thinking about things for this PR but I would consider defaulting the felixConfiguration HealthPort and writing that back to the datastore.
Do you see any issue with doing that?

**Comment by howels** [comment]
Are you able to incorporate a similar change for Typha Healthport also?

**Comment by caseydavenport** [comment]
> Are you able to incorporate a similar change for Typha Healthport also?

There's no datastore config for this at the moment, so I can't do the exact same thing. However, I can probably make it so that Typha uses a port based on the felix port (e.g., typhaPort = felixPort + 1)

**Comment by howels** [comment]
> > Are you able to incorporate a similar change for Typha Healthport also?
> 
> There's no datastore config for this at the moment, so I can't do the exact same thing. However, I can probably make it so that Typha uses a port based on the felix port (e.g., typhaPort = felixPort + 1)

If you are happy to do so, that is a step in the right direction and would help us to avoid clashes with existing applications.

**Comment by caseydavenport** [comment]
Pushed a new commit that sets the typha port to be `felixPort -1`, which matches the old default behavior. 

@tmjd I think this is ready for a review.

**Review by tmjd — APPROVED** [review]
LGTM, a couple notes though.
So this is actually going to be a change for OpenShift clusters I think, Typha will now be changing from 9098 to 9198. Hopefully not a big deal but something to be aware of (please correct me if I missed something and that is not the case).

Also I'm not sure I like the `-1` for typha happening in the typha render, I think it would be more appropriate in the controller but not enough to block this PR, we can always change it later.

**Comment by howels** [comment]
Thanks for doing this work. When is the next release planned now that this is included @caseydavenport ?

**Pull request merged**